src/main/java/seedu/sellsavvy/logic/commands/ordercommands/MarkOrderCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.sellsavvy.logic.commands.ordercommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.logic.Messages.MESSAGE_ORDERLIST_DOES_NOT_EXIST;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
+import seedu.sellsavvy.logic.commands.Command;
+import seedu.sellsavvy.logic.commands.CommandResult;
+import seedu.sellsavvy.logic.commands.exceptions.CommandException;
+import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Count;
+import seedu.sellsavvy.model.order.Date;
+import seedu.sellsavvy.model.order.Item;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.order.Status;
+import seedu.sellsavvy.model.person.Person;
+
+/**
+ * Marks an order as completed.
+ */
+public class MarkOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "markOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks an order as completed."
+            + "Parameters: ORDER_INDEX (must be positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_MARK_ORDER_SUCCESS = "The order has been marked as completed: %1$s";
+
+    private final Index index;
+
+    /**
+     * Creates a MarkOrderCommand to mark the order under the
+     * specified index as completed.
+     *
+     * @param index of the order in the displayed order list to mark.
+     */
+    public MarkOrderCommand(Index index) {
+        requireNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Person selectedPerson = model.getSelectedPerson().get();
+        if (selectedPerson == null) {
+            throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
+        }
+
+        OrderList orderList = selectedPerson.getOrderList();
+        if (index.getZeroBased() >= orderList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+        }
+        Order orderToMark = orderList.get(index.getZeroBased());
+        Order newOrder = createMarkedOrder(orderToMark);
+
+        orderList.setOrder(orderToMark, newOrder);
+        return new CommandResult(String.format(MESSAGE_MARK_ORDER_SUCCESS, Messages.format(newOrder)));
+    }
+
+    /**
+     * Creates a marked version of the given {@code Order}.
+     */
+    public static Order createMarkedOrder(Order order) {
+        assert order != null;
+
+        Item item = order.getItem();
+        Date date = order.getDate();
+        Count count = order.getCount();
+        Status status = Status.COMPLETED;
```

Comment:
Not sure if it is actually important but have u considered the case where the order is already completed

Reply:
have considered but ultimately the result is the same, it's like editing a Person with name "Alice" to "Alice". Would you have preferred if a message is displayed saying "The order has already been marked as completed"?

Reply:
I think that would been better because if a user mark something that is already completed it is likely that it is a typo plus it may cause unnecessary re-rendering of the gui if we replace stuff unnecessarily in the orderList.

Reply:
will create a new check under the `MarkOrderCommand::execute` method which throws a `CommandException` if the order is already marked as completed.